react/components/admin/redirects/UploadModal/UploadPrompt/index.tsx

```diff
@@ -0,0 +1,295 @@
+import { UploadActionType } from '../../mutations/SaveRedirectFromFile'
+import { CSV_HEADER, CSV_SEPARATOR } from '../../consts'
+import { isRedirectType, validateRedirect } from './validateRedirect'
+import styles from './UploadPrompt.css'
+
+interface Props {
+  hasRedirects: boolean
+  onClickBack: () => void
+  onDownloadTemplate: () => void
+  saveRedirectFromFile: (
+    parsedJsonFromCsv: Redirect[],
+    fileName: string
+  ) => void
+  uploadActionType: UploadActionType
+}
+
+const messages = defineMessages({
+  alertSeeErrors: {
+    defaultMessage: 'See errors',
+    id:
+      'admin/pages.admin.redirects.upload-modal.prompt.alert.validation-errors.action',
+  },
+  body: {
+    defaultMessage: 'Please use our template or an exported file as base.',
+    id: 'admin/pages.admin.redirects.upload-modal.prompt.body',
+  },
+  buttonCancel: {
+    defaultMessage: 'Cancel',
+    id: 'admin/pages.admin.redirects.upload-modal.prompt.button.cancel',
+  },
+  buttonImport: {
+    defaultMessage: 'Import file',
+    id: 'admin/pages.admin.redirects.upload-modal.prompt.button.import',
+  },
+  delete: {
+    defaultMessage: 'Delete Redirects',
+    id: 'admin/pages.admin.redirects.upload-modal.file.title-delete',
+  },
+  dropButton: {
+    defaultMessage: 'choose a file',
+    id: 'admin/pages.admin.redirects.upload-modal.prompt.drop.button',
+  },
+  dropMessage: {
+    defaultMessage: 'Drop here your CSV or ',
+    id: 'admin/pages.admin.redirects.upload-modal.prompt.drop.message',
+  },
+  downloadTemplate: {
+    defaultMessage: 'Download template',
+    id:
+      'admin/pages.admin.redirects.upload-modal.prompt.body.download-template',
+  },
+  save: {
+    defaultMessage: 'Save Redirects',
+    id: 'admin/pages.admin.redirects.upload-modal.file.title-save',
+  },
+  unsupportedFileFormat: {
+    defaultMessage: 'Unsupported file type. Please upload a .csv file.',
+    id:
+      'admin/pages.admin.redirects.upload-modal.prompt.alert.unsupported-file',
+  },
+  validationErrors: {
+    defaultMessage: 'Validation errors on file {name}',
+    id:
+      'admin/pages.admin.redirects.upload-modal.prompt.alert.validation-errors',
+  },
+})
+
+interface AlertState {
+  message: string
+  action?: AlertProps['action']
+}
+
+const UploadPrompt: React.FC<Props> = ({
+  hasRedirects,
+  onClickBack,
+  onDownloadTemplate,
+  saveRedirectFromFile,
+  uploadActionType,
+}) => {
+  const intl = useIntl()
+  const [file, setFile] = useState<File | null>(null)
+  const [shouldShowErrors, setShouldShowError] = useState(false)
+  const [parsedJsonFromCsv, setParsedJsonFromCsv] = useState<
+    Redirect[] | undefined
+  >()
+
+  const [alertState, setAlert] = useState<AlertState | undefined>()
+  const [validationErrors, setValidationErrors] = useState<
+    string[] | undefined
+  >()
+
+  const resetErrors = useCallback(() => {
+    setShouldShowError(false)
+    setValidationErrors(undefined)
+    setAlert(undefined)
```

Comment:
:pick: Are those `undefined` necessary?

Reply:
I could set the initial state value as `null` instead of `undefined`, maybe it's clearer.

Reply:
I've changed it to `null` to make it more readable. Although the `undefined` is unnecessary I think having `setAlert(null)` is better to read than `setAlert()`.